init/main.go

```diff
@@ -619,6 +622,15 @@ func boost() error {
 		return err
 	}
 
+	// Mount efivarfs if running in EFI mode
+	if _, err := os.Stat("/sys/firmware/efi"); errors.Is(err, os.ErrNotExist) {
+		if err := mount("efivarfs", "/sys/firmware/efi/efivars", "efivarfs", unix.MS_NOSUID|unix.MS_NOEXEC|unix.MS_NODEV, ""); err != nil {
```

Comment:
See`moveMountpointsToHost()` function. First it tries to move the mountpoint to host. If it does not work (e.g. new root does not contain the directory) then it unmounts the filesystem from initram fs tree.

The `moveMountpointsToHost()` should handle `/sys/firmware/efi` mountpoint as well. I wonder if `MS_MOVE` and `MNT_DETACH` handle mountpoints recursively.

Reply:
I've been reading through the documentation and I think that moving the mountpoint isn't needed at all. It looks like systemd handles all the mountpoints but `/run`.  Take a look at the [specification](https://systemd.io/INITRD_INTERFACE/), it may even be the case that the current code on `moveMountpointsToHost()` is mostly unneeded.

Reply:
It actually used to be like mentioned in this specification. But it turns out that other init systems like `runit` expect `/dev/` to be mounted at the boot time. See https://github.com/anatol/booster/issues/92 and https://github.com/anatol/booster/commit/4c9505873ec5ecde77ed749091ce7bd106b5b401

I am fine to leave this code as-is and wait for integration tests results to see if runit tests pass.

Reply:
Then I'll try to get unit tests ready today and see if it works.

Reply:
I ran your PR with VoidLinux and ArchLinux+btrfs tests and they passed.